### PR TITLE
Replace tabs by Navigation-Drawer

### PR DESCRIPTION
**Pull request by ligi**
device-2015-10-16-212026

- Brings back the logo thanks to @cbalster for the gfx support
- more screen-real-estate in the map-view
- prepares usage of load-toast ( would otherwise clash with tabs )
- nicer ui and less code ( +146 −170 )

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -44,7 +44,8 @@ dependencies {
     compile 'com.android.support:design:23.1.0'
     compile 'com.squareup.okhttp:okhttp:2.5.0'
     compile 'com.jakewharton:butterknife:7.0.1'
-    
+    compile 'org.ligi:AXT:0.35'
```
Comment (cbalster) [inline app/build.gradle]: I know this is your baby and from quickly looking at it it's got a lot of nice stuff. But IMHO and more as a general point I think we should carefully consider if saving a few lines of code (13 in this case) ist worth adding another dependency..

File: app/src/main/java/de/stephanlindauer/criticalmaps/Main.java [hunk 2]
```diff
@@ -56,6 +70,15 @@ public void onCreate(Bundle bundle) {
         locationUpdatesService.initialize(getApplication());
 
         startSyncService();
+
+        drawerNavigation.setNavigationItemSelectedListener(this);
+
+        setSupportActionBar(toolbar);
+        ActionBarDrawerToggle mDrawerToggle = new ActionBarDrawerToggle(this, drawerLayout, toolbar, R.string.open, R.string.close);
```
Comment (cbalster) [inline app/src/main/java/de/stephanlindauer/criticalmaps/Main.java]: those strings are for accessibility so especially the close string makes no sense here. I suggest something like "open drawer" and "close drawer"

File: app/src/main/java/de/stephanlindauer/criticalmaps/Main.java [hunk 3]
```diff
@@ -56,6 +70,15 @@ public void onCreate(Bundle bundle) {
...
+        drawerLayout.setDrawerListener(mDrawerToggle);
+        mDrawerToggle.syncState();
```
Comment (cbalster) [inline app/src/main/java/de/stephanlindauer/criticalmaps/Main.java]: the docs state this should be called in onPostCreate(). Did you look into that?

**Comment by cbalster** [comment]
Could you please check your indentation in the xmls again - they don't conform to the default Android Studio settings. Just to prevent from new commits from unneccesarily changing that again.
Besides my line notes everything else looks good :+1: 
Nice :)

**Comment by ligi** [comment]
Thanks for the comments - just addressed all of them except removing AXT - sure it just saves 13 lines at the moment - but a) every line that is not in your main project is a good one b) the lib is really small and only jar not aar c) might reduce more code in the future - just do not want to let this PR get even bigger

**Comment by ligi** [comment]
would be great if @stephanlindauer and @johnjohndoe could have a look at this change as it is a functional-change this time

**Comment by stephanlindauer** [comment]
yep yep yep. i like it. a lot. but i cant have a closer look right now. will do on monday.

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -126,5 +126,6 @@
     <string name="introduction_careful">Please rather pay attention to traffic instead of the app.</string>
     <string name="introduction_bugs">If you find bugs in the app, please let us know by clicking the feedback-button in the settings-menu.</string>
     <string name="introduction_requisites">In order for this app to run properly, you need to have a working mobile data connection and GPS enabled.</string>
-
+    <string name="open_drawer">open drawer</string>
```
Comment (johnjohndoe) [inline app/src/main/res/values/strings.xml]: I would capitalize the first word. Same with `close_drawer` text.

**Comment by johnjohndoe** [comment]
Same here. Can check this on Monday the earliest.

**Comment by johnjohndoe** [comment]
@ligi Two improvements:

1. When the user the selects a menu item in the navigation drawer which points to the fragment which is currently inflated then the navigation drawer should just be closed. There is no need to inflate the fragment again.
2. The navigation drawer menu items should visually reflect which menu item is selected (the currently inflated fragment).

**Comment by ligi** [comment]
@johnjohndoe implemented both suggestions - still separated commits for your reviewing pleasure - will squash before merge

**Comment by ligi** [comment]
device-2015-10-19-222047

**Comment by cbalster** [comment]
:+1: Think this is good to. Any other issues/improvements can be done later in separate patches.

**Comment by stephanlindauer** [comment]
just one suggestion: what if we have users with tiny screens? do we hide the logo? should we put the menu items inside a scroll view?

**Comment by cbalster** [comment]
@johnjohndoe yes, it is - it's set enabled in the xml

**Comment by cbalster** [comment]
@stephanlindauer Screenshot from 480x800 emulator

device-2015-10-20-104435

**Comment by stephanlindauer** [comment]
another suggestion: most people are right-handed, therefore taping with their right thumb. wouldn't it make sense to align those items to the right of the (still left aligned) nav drawer?
also: how about applying a opaque background so the user can still see whats happening in the fragment behind the drawer?
sorry i am just talk and no actions. telekom moved my appointment back another week. :(

**Comment by cbalster** [comment]
Items are tappable over the whole width of the drawer anyway, so I don't
think this is necessary.
Oh an I'm right handed, but still use my phone with the left most of the
time :)

**Comment by johnjohndoe** [comment]
@stephanlindauer Textual input is always welcome.

---

Besides my comments regarding the implementation I am personally not sure that the drawer is the best pattern for the user experience for the use case of the application. When I think of people riding their bike, swiping to the drawer and pointing at a menu item appears to me not the most convenient thing. Car navigation apps have big buttons which are accessible right away for reasons. We should think about this.

**Comment by ligi** [comment]
@johnjohndoe I think you lost your ability to veto against this feature as a whole when suggesting detail changes that I already did now ;-)
The big button is be the Burger in the toolbar - also the idea is that when biking the user should interact as little as possible with the app - and see as much as possible from the map.
We do not ( yet ) have measurements - but I think in the use-case of participating in the cm the user is mainly on the map with some peeks in twitter and chat. My Idea would be that chat and twitter-messages appear as snackbars on the app to minimize needed user-interaction

**Comment by cbalster** [comment]
Okay, so I also had concerns regarding the navdrawer due to the "2 actions vs. 1 action" issue to change views. But since in my personal use of the app I basically only look at the map, I think that's an okay tradeof for having a bigger map. -> I'm for merging it.

**Comment by johnjohndoe** [comment]
I am okay with the change after @ligi explained his intentions.

**Comment by cbalster** [comment]
yay, concensus :)
@stephanlindauer are you okay with merging it too?
@ligi do you still want to squash those last two commits in?

**Comment by stephanlindauer** [comment]
yep. go for it!

**Comment by ligi** [comment]
great! Commits are squashed now

**Pull request merged**